src/gateway/router/routes/contractDataRoute.ts

```diff
@@ -86,21 +89,29 @@ async function getContractData(
 
   logger.debug('strData', strData);
 
-  const tx = new Transaction({ ...JSON.parse(strData) });
-  const txData = Buffer.from(tx.data);
-
-  // get contract transaction content type from its tag
-  const contentType = getContentType(tx);
-  logger.debug(`Content type for id: ${id}: `, contentType);
-
-  return { data: txData, contentType };
+  if (deploymentType == 'warp-external') {
```

Comment:
extract to a type and describe its meaning.

Reply:
done